src/Microsoft.OpenApi.OData.Reader/Operation/EntityDeleteOperationHandler.cs

```diff
@@ -81,7 +81,11 @@ protected override void SetParameters(OpenApiOperation operation)
         /// <inheritdoc/>
         protected override void SetResponses(OpenApiOperation operation)
         {
-            operation.AddErrorResponses(Context.Settings, true);
+            // Response for Delete methods should be 204 No Content
+            OpenApiConvertSettings settings = Context.Settings.Clone();
```

Comment:
I'm a bit late to the party here. But if we could avoid allocs for that kind of thing, it'd be better. Could we not check the operation method/verb in the AddErrorResponses method instead? @irvinesunday

Reply:
I thought about it, but I was avoiding growing this function by adding an extra optional parameter (which will only be utilized by delete methods) and conditional checks here. I also wanted to be explicit that we are not using success status code range for `delete` methods.

Reply:
thanks for the additional context here. Would it have been another parameter since it's an extension method on operation and thus it has access to the operation properties?

Reply:
But you can't infer the `OperationType` from the `operation` object properties. This is in the `Operations` property dictionary key of an `OpenApiPathItem` object: https://github.com/microsoft/OpenAPI.NET/blob/df1b405e537d0f3c2a829b6a80fef82959d0eab5/src/Microsoft.OpenApi/Models/OpenApiPathItem.cs#L29
`Operation` properties: https://github.com/microsoft/OpenAPI.NET/blob/df1b405e537d0f3c2a829b6a80fef82959d0eab5/src/Microsoft.OpenApi/Models/OpenApiOperation.cs#L119-L131

Reply:
ah right, I forgot about the dictionary pattern, which is a bit painful to carry the context around. Fine, leave it as is for now, thanks for taking the time to discuss this.

Reply:
No worries.